src/mscorlib/shared/System/Guid.Unix.cs

```diff
@@ -10,11 +10,12 @@ namespace System
     partial struct Guid
     {
         // This will create a new random guid based on the https://www.ietf.org/rfc/rfc4122.txt 
-        public static unsafe Guid NewGuid()
+        public static Guid NewGuid()
         {
-            Guid g;
-            Interop.GetRandomBytes((byte*)&g, sizeof(Guid));
-            
+            Span<byte> randomBytes = stackalloc byte[Marshal.SizeOf<Guid>()];
```

Comment:
To remove unsafe i can't use `sizeof(Guid)` anymore, so i followed compiler advice and used `Marshal.SizeOf<Guid>()`. Is better to use `Marshal.SizeOf<Guid>()`(internal call) or new `Unsafe.SizeOf<Guid>()`(IL opcode)?

Reply:
We try to avoid using `Marshal.SizeOf` as much as possible. It is slow expensive operation. This change is likely introducing a pretty significant performance regression.

Do not worry about whether the code has `unsafe` keyword or not. The `unsafe` keyword is sort of a joke. It does not say much about whether the code is actually unsafe. It is required on code that is perfectly safe, like `sizeof`. On the other hand, it is not required on code that is pretty unsafe, like passing pointers via `ref`.

Reply:
@jkotas do you have a code pointer(on codebase) where CoreClr do IL verification(maybe you save me some time)?

Reply:
CoreCLR does not do IL verification. It is always full trust.

We have a standalone IL verifier in CoreRT repo: https://github.com/dotnet/corert/tree/master/src/ILVerify